Add functionality to update Auth0 users

gene1wood opened the pull request:
With this the Auth0 user is updated via the Auth0 management API

gene1wood committed:
Add documentation and digrams

gene1wood committed:
Add functionality to update Auth0 users

With this the Auth0 user is updated via the Auth0 management API

fiji-flo commented on functions/auth0_cis_webhook_consumer/utils.py:
```diff
@@ -36,69 +47,197 @@ def verify_token(authorization: str) -> bool:
+            if data.get('values') is None:
+                continue
+            for value in data['values']:
+                if data['values'][value] is None:
```
We use `""` as value for pmo managed mozilliansorg groups. Any particular reason for this check, or can we just omit it?

gene1wood replied:
@fiji-flo Ah! That's great to know.

Yes, the purpose of this check is to filter out content from the `hris` section of the profile

```json
{
    "access_information": {
        "hris": {
            "values": {
                "egencia_pos_country": "US",
                "employee_id": "1234",
                "managers_primary_work_email": "jdoe@mozilla.com",
                "primary_work_email": "jdoeboss@mozilla.com",
                "worker_type": "Employee"
            }
        }
    }
}
```

I didn't realize that there would be non `null` values in the `mozilliansorg` section. I don't see any in my and April's profile but maybe others have it? Anyhow, I'll change this check.

gene1wood replied:
@fiji-flo How does [this look](https://github.com/mozilla-iam/auth0-cis-webhook-consumer/pull/3/commits/ffc7205c3aba7bae85ad635220e4b950cc87b33b) as a solution?

gene1wood committed:
Change logic that seeks groups in profile

mozilliansorg groups can sometimes have a `''` value instead of a `null` value. This change
supports profiles with fields of that type by no longer considering the value in deciding
whether or not to integrate the group into the Auth0 profile

gene1wood committed:
Add user whitelist feature

This will allow for deploying this live but only affecting a subset of users. This is to
validate that it works correctly in production before cutting everyone over.
